readthedocs/projects/views/private.py

```diff
@@ -213,6 +214,33 @@ def form_valid(self, form):
         return HttpResponseRedirect(self.get_success_url())
 
 
+class ProjectVersionsSync(ProjectVersionMixin, GenericView):
+
+    """
+    Re-sync a project repository.
+
+    The signal will add a success/failure message on the request.
+    """
+
+    def post(self, request, *args, **kwargs):
+        # pylint: disable=unused-argument
+        project = self.get_project()
+        sync = sync_versions(project)
+
+        if sync:
+            messages.success(
+                request,
+                _('Project version sync triggered'),
+            )
+        else:
+            messages.error(
+                request,
+                _("There was an error when syncing project's versions"),
```

Comment:
Same feedback here, but also: there isn't much the user can do with this error. Do we have any additional information we can display in the `sync` variable? What error does a failure with the sync task point at otherwise? Perhaps there is more guidance that we can give the user.

As for the message, it's better to speak to the user -- something closer to:

```suggestion
                _("There was a problem updating the project versions"),
```

Reply:
@agjohnson Actually `sync_versions` function only calls a celery task `sync_repository_task` So, I don't know if we can show much more information from `sync` variable.